10_rest/runnerdave/pkg/puppy/rest.go

```diff
@@ -0,0 +1,90 @@
+package puppy
+
+import (
+	"encoding/json"
+	"log"
+	"net/http"
+	"strconv"
+
+	chi "github.com/go-chi/chi"
+)
+
+// RestServer store for puppies
+type RestServer struct {
+	DB Storer
+}
+
+// SetupRoutes setups the routes for the server
+func (rs *RestServer) SetupRoutes(db Storer) *chi.Mux {
+	r := chi.NewRouter()
+	rs.DB = db
+
+	r.Route("/api/puppy", func(r chi.Router) {
+		r.Post("/", rs.createPuppy)
+		r.Route("/{id}", func(r chi.Router) {
+			r.Put("/", rs.updatePuppy)
+			r.Delete("/", rs.deletePuppy)
+			r.Get("/", rs.getPuppy)
+		})
+	})
+
+	return r
+}
+
+func (rs *RestServer) getPuppy(w http.ResponseWriter, r *http.Request) {
+	id, err := strconv.Atoi(chi.URLParam(r, "id"))
+	if err != nil {
+		w.WriteHeader(http.StatusBadRequest)
+		return
+	}
+	p, err := rs.DB.ReadPuppy(int16(id))
+	if err != nil {
+		w.WriteHeader(http.StatusNotFound)
+		return
+	}
+	w.Header().Set("Content-Type", "application/json")
+	bytes, _ := json.Marshal(p)
+	i, _ := w.Write(bytes)
+	log.Printf("Content length:%v", i)
+}
+
+func (rs *RestServer) createPuppy(w http.ResponseWriter, r *http.Request) {
+	p := Puppy{}
+	if err := json.NewDecoder(r.Body).Decode(&p); err != nil {
+		w.WriteHeader(http.StatusBadRequest)
+		return
+	}
+	if err := rs.DB.CreatePuppy(p); err != nil {
+		w.WriteHeader(http.StatusInternalServerError)
+		return
+	}
+}
```

Comment:
According to REST, this should probably return a 201 Created status here and not the default 200 OK.

Reply:
done